### PR TITLE
Connected peer list

### PR DESCRIPTION
**Pull request by mplorentz**
#254

This adds UI in `MenuViewController` to display a list of connected peers and the number of messages that have recently been downloaded. On top of that this contains:
- The first SwiftUI views in Planetary
- The first use of the Combine framework
- The second use of Swift's new structured concurrency support aka `async/await`
- The first try at a IDD architecture in the view layer
- My first interaction with our localization scheme

As such there are a lot of extensions and infrastructure code I had to write in addition to just making the new view. There are also places where I took some shortcuts because this took way longer than I had planned. I'd love to hear feedback on the View/Coordinator/Router/Service architecture I used. I don't think it's a perfect model but I figured we could start with this and evolve the approach as we go.

I also filed #404 so we can start generating colors for SwiftUI. Right now I am instantiating them with strings which isn't great.

### CODE REVIEW

File: Source/Controller/ConnectedPeerListCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+                
+                peerConnectionInfo.append(PeerConnectionInfo(about: about))
+            } catch {
+                Log.optional(error)
+            }
+        }
+        
+        return peerConnectionInfo.sorted { lhs, rhs in
+            guard lhs.isActive == rhs.isActive else {
+                return lhs.isActive ? true : false
```
Comment (martindsq) [inline Source/Controller/ConnectedPeerListCoordinator.swift:176]: return lhs.isActive directly

File: Source/Controller/ConnectedPeerListCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    
+    // MARK: - User Actions
+    
+    func peerTapped(_ connectionInfo: PeerConnectionInfo) {
+        guard let identity = connectionInfo.identity else {
+            router.alert(error: ConnectedPeerListError.identityNotFound)
+            return
+        }
+        
+        router.showProfile(for: identity)
```
Comment (martindsq) [inline Source/Controller/ConnectedPeerListCoordinator.swift:134]: Might want to track this event in analytics with trackDidTapButton or similar

File: Source/Localization/en.lproj/Generated.strings [hunk 3]
```diff
@@ -18,6 +18,7 @@
 "Text.future" = "In the future";
 "Text.daysAgo" = "{{ days }} days ago";
 "Text.atDayTime" = "{{ day }} at {{ time }}";
+"Text.minutesAbbreviated" = "{{ numberOfMinutes }} mins";
```
Comment (martindsq) [inline Source/Localization/en.lproj/Generated.strings:21]: We can add the Spanish and Polish translations, I can obviously give a Spanish translation. But I think that Sebastian really like to translate texts. Maybe we can create a channel in slack where we can past texts in English and ask the rest to give translations.

Comment (mplorentz) [reply Source/Localization/en.lproj/Generated.strings:21]: Filed #408

File: Shared/Extensions/UIViewController+Alert.swift [hunk 4]
```diff
@@ -10,7 +10,15 @@ import Foundation
...
+    
+    /// Presents an error message to the user in an alert box.
+    /// - Parameter error: The error to present to the user. The `localizedDescription` is used for the error message.
+    func alert(error: Error)
+}
+
+extension UIViewController: AlertRouter {
     
     /// Presents an error message to the user in an alert box.
     /// - Parameter error: The error to present to the user. The `localizedDescription` is used for the error message.
```
[the remark this answers is not in the document]
Comment (mplorentz) [reply Shared/Extensions/UIViewController+Alert.swift:24]: 👍thanks, I went ahead and removed it here.

File: Source/Controller/ConnectedPeerListCoordinator.swift (hunk 1 above)
Comment (mplorentz) [reply Source/Controller/ConnectedPeerListCoordinator.swift:176]: 🤦‍♂️😂

**Review by martindsq — APPROVED** [review]

**Pull request merged**